server/config/db.js

```diff
@@ -0,0 +1,15 @@
+const mongoose = require('mongoose');
+
+mongoose.Promise = global.Promise;
```

Comment:
Mongoose native library may be deprecated (as of February)

according to the docs, you need another line 
 ``` javascript
// Use native promises
    mongoose.Promise = global.Promise;
    assert.equal(query.exec().constructor, global.Promise);
``` 


http://mongoosejs.com/docs/promises.html